### ISSUE TITLE
One click option to add flathub repo.

### ISSUE THREAD

**Issue by Crono23** [elementary/appcenter#1748]
### Problem

Having very few apps by default can be misleading to some. Opening up flathub through browser also takes a little bit of extra time.

### Proposal

It would be a nicer experience if we could add flathub repo in one click from the AppCenter.

### Prior Art (Optional)

In a clean Fedora installation, the Gnome software center asks the user whether or not they want to enable a third-party repos. Recently they added flathub as one of the repo, albeit limited by a filter. This is all done without having to open up a web browser.

Here's their related documentation :
[Filtered flathub apps](https://fedoraproject.org/wiki/Changes/Filtered_Flathub_Applications)
[Third-party software mechanism](https://fedoraproject.org/wiki/Changes/Third_Party_Software_Mechanism)

**Comment by danrabbit** [comment]
TL;DR the friction here is intentional, so this is a design conflict

FlatHub has a couple of big problems that prevent us from wanting to ship it by default:
* There are no guarantees a Flathub app is developed or distributed by who it says it is
* There are no human reviews of updates to apps on Flathub, including updates that further open the sandbox

The above are very important because there's no way to completely sandbox apps on X11 which means that generally we don't consider this a completely safe software source. It also means that there are several apps which are redistributed without the original authors knowledge or endorsement which leads to them not being well tested with regards to sandboxing or even having accurate appdata. Not having accurate appdata is a big problem because it means users can't have informed consent with regards to app content when they install. This is one of the promises we want to make with AppCenter and shipping Flathub by default would break that promise.

We learned from shipping the Ubuntu repos under the "uncurated" header and with a warning dialog that
* People hate warning dialogs, going so far as to call them malicious and prescribe ill intent to elementary over them
* People don't read the warning dialogs, meaning they don't actually do their job of warning about the potential pitfalls people may face
* When things go wrong, people blame elementary and AppCenter for things that are shipped by default.

So it's necessary to have some friction when adding 3rd party sources to manage expectations and ensure the ootb experience is safe. Making sure that people explicitly go outside of appcenter to add 3rd party apps and sources makes it more clear that these apps are not something elementary claims responsibility over.

---

Filters aren't really an acceptable solution because it conflicts with the goals and intentions of elementary OS as a software platform. elementary OS does not follow the Linux distribution model in that elementary does not package and redistribute third party software (other than what's pre-installed). Maintaining a filter here would be functionally equivalent to becoming software maintainers, which is not something we're interested in doing. Instead elementary OS follows an app store model where app developers who wish to publish for elementary OS can do so through AppCenter.

---

I also want to note some personal concerns with showing too much preferential treatment towards Flathub. One of the reasons elementary chose Flatpak as a technology specifically was for decentralization. I think it's important that the direction we give users is that Flatpak as a technology enables sideloading from any number of 3rd party software sources and not imply that Flathub is the One True Way to sideload apps. We already direct users specifically to Flathub during onboarding and when AppCenter search is empty. This is already potentially problematic towards this end in my personal opinion.

**Comment by Letterus** [comment]
All thumbs up for a curated and maintained Flatpak repo and not leading users to anything like Flathub automatically.

I use Flathub, but in addition to the technical issues @danrabbit mentioned I have a very bad feeling about it concerning some further issues:

1. Who runs it? I'm from Europe and I'm missing an imprint or any legal notice that informs users about who is responsible for the site and operation (who controlls and checks actions/processes and data/apps?). That's stuff we expect from serious software distributors overs here. ;-)
2. Who maintains the runtimes? How do I know security issues are fixed? Who is responsible for that?
3. Same about data privacy: Which information is stored and evaluated when using the site and the repo? I do not detect any information about responsibilites requested by GDPR of European law (https://gdpr.eu/).

So, I use it with my dev machine to develop for elementary OS, but I won't let it run on anything meant for production. For users there is now way I may recommend using Flathub as long as . So I think elementary OS should not as well.

**Comment by prcastro** [comment]
Having an empty AppCenter is one of the main issues with the distro right now. This was pointed out by MakeUseOf, InfinitelyGalactic, DistroTube, Linux Experiment, The Register and many (many) comments from users.

While I truly understand the rationale behind not adding Flathub, I'd like to propose a compromise: package and distribute flatpaks for top applications on AppCenter. I'm talking about:

* Firefox
* Chromium
* GIMP
* Libre office
* VLC
* Spotify
* Discord
* Steam

and maybe a few more

While many of those won't respect the look & feel of the OS, the benefits are tremendous. Those are applications many people want and need to perform simple tasks on the desktop.

With this, even Linus Tech Tips would easily use elementary as their daily driver ;)

**Comment by Letterus** [comment]
I don't think that's a good strategy or in line with the project's principles.

Workaround:

```sudo apt install synaptic```
-> use synaptic and install whatever you like to

**Comment by prcastro** [comment]
That's really unfortunate.

I'm not sure how much that's worth, but my counter argument would be that this approach goes against the elementary HIG:

> Required Configuration
> When a user first launches an app, they should be able to get down to business as quickly as possible. If configuration is not absolutely required for the first use, they should not be required to configure anything. If configuration is required, they should be presented with a clean and simple welcome screen within the app. Avoid separate configuration dialogs when launching.

If "business" means editing office files and photos, gaming or listening to music, they probably will need to configure stuff AND wouldn't be greeted with a welcome screen to configure those apps.

Not to mention the fact that the recommended solution is to use the command line.

But hey, again, not my personal project nor my vision, just pointing out things I missed as a user.

**Comment by getsnoopy** [comment]
@prcastro Of course, the irony is that if they were following that principle, they wouldn't be defaulting to the *English (US)* locale either or pushing that locale subtly throughout the OS, seeing as it's used by a mere 26% of the world's English-speaking population. The majority of English users end up having to configure elementary to work properly for them anyway.

While I really like elementary OS, as I think it's the best designed Linux OS out there, it's unfortunate that the team is making poor decisions such as these to make it more difficult for new users to use the platform, ultimately working against the purpose of their entire distro to begin with.

**Comment by Sampo-Kolu** [comment]
For anyone in the future looking at this: It was added in OS 8.0.
